src/client/Microsoft.Identity.Client/Internal/Requests/ManagedIdentityAuthRequest.cs

```diff
@@ -31,159 +32,278 @@ public ManagedIdentityAuthRequest(
 
-            // If we have claims, we do NOT use the cached token (but we still need it to compute the hash).
-            if (!string.IsNullOrEmpty(AuthenticationRequestParameters.Claims))
+            if (hasClaims)
             {
-                _managedIdentityParameters.Claims = AuthenticationRequestParameters.Claims;
+                if (string.IsNullOrEmpty(_managedIdentityParameters.Claims))
+                {
+                    _managedIdentityParameters.Claims = AuthenticationRequestParameters.Claims;
+                }
+
                 AuthenticationRequestParameters.RequestContext.ApiEvent.CacheInfo = CacheRefreshReason.ForceRefreshOrClaims;
 
-                // If there is a cached token, compute its hash for the “revoked token” scenario
                 if (cachedAccessTokenItem != null)
                 {
-                    string cachedTokenHash = _cryptoManager.CreateSha256HashHex(cachedAccessTokenItem.Secret);
+                    // Compute revoked‑token hash from the cached token
+                    string cachedTokenHash = ServiceBundle.PlatformProxy.CryptographyManager.CreateSha256HashHex(cachedAccessTokenItem.Secret);
                     _managedIdentityParameters.RevokedTokenHash = cachedTokenHash;
 
-                    logger.Info("[ManagedIdentityRequest] Claims are present. Computed hash of the cached (revoked) token. " +
-                                "Will now request a fresh token from the MI endpoint.");
+                    logger.Info("[ManagedIdentityRequest] Claims present. Computed hash of the cached (revoked) token. Will request a fresh token.");
                 }
                 else
                 {
-                    logger.Info("[ManagedIdentityRequest] Claims are present, but no cached token was found. " +
-                                "Requesting a fresh token from the MI endpoint without a revoked-token hash.");
+                    logger.Info("[ManagedIdentityRequest] Claims present but no cached token found. Requesting a fresh token without revoked-token hash.");
                 }
 
-                // In both cases, we skip using the cached token and get a new one
-                authResult = await GetAccessTokenAsync(cancellationToken, logger).ConfigureAwait(false);
-                return authResult;
+                return await AcquireFreshTokenAsync(
+                    CacheRefreshReason.ForceRefreshOrClaims,
+                    "[ManagedIdentityRequest] Claims provided; bypassing cache.",
+                    popRequested,
+                    cancellationToken,
+                    logger).ConfigureAwait(false);
             }
 
-            // 3. If we have no ForceRefresh and no claims, we can use the cache
-            if (cachedAccessTokenItem != null)
+            // 4) If PoP requested but no binding cert has been applied yet, bypass cache and mint one
+            if (popRequested && AuthenticationRequestParameters.AuthenticationOperationOverride == null)
```

Comment:
Checking this property means is a convoluted way of saying "do I have a cached cert?". Can you check the cert instead?